### ISSUE TITLE
Test.py does not work with Python 3

### ISSUE THREAD

**Issue by MarkusTeufelberger** [ripple/rippled#2272]
I'll happily commit the fix (dicts don't have iteritems() in Python 3), but I'd also like to ask if it would be ok to do some minor refactoring there, especially around the cmake targets. Now looking at the code again after a while, it seems to be more confusing that in probably should be, considering that it is not very performance critical, but it is important to know what's going on in there.

**Comment by MarkusTeufelberger** [comment]
I am actually not able to run the line `CMAKE_UNITY_COMBOS = { '' : [['rippled', 'rippled_classic'], CMAKE_UNITY_CONFIGS],
    '.nounity' : [['rippled', 'rippled_unity'], CMAKE_NONUNITY_CONFIGS] }` at all - could someone please just give me the contents of `CMAKE_DIR_TARGETS` on a Linux machine?

**Comment by MarkusTeufelberger** [comment]
Ok, tried around a bit more, apparently it contains a data structure with magic values that are just expected to be there in a dict with no real names for keys etc.

I'll refactor this one, even if for my own sanity. :-/

**Comment by ximinez** [comment]
With Python 2.7.14
```
CMAKE_UNITY_COMBOS: {u'': [[u'rippled', u'rippled_classic'], []], u'.nounity': [[u'rippled', u'rippled_unity'], []]}

CMAKE_DIR_TARGETS: {(u'gcc.release',): [[u'rippled', u'rippled_classic'], []], (u'clang.coverage',): [[u'rippled', u'rippled_classic'], []], (u'gcc.debug',): [[u'rippled', u'rippled_classic'], []], (u'gcc.coverage',): [[u'rippled', u'rippled_classic'], []], (u'gcc.debug.nounity',): [[u'rippled', u'rippled_unity'], []], (u'gcc.profile',): [[u'rippled', u'rippled_classic'], []], (u'clang.profile',): [[u'rippled', u'rippled_classic'], []], (u'gcc.coverage.nounity',): [[u'rippled', u'rippled_unity'], []], (u'clang.release',): [[u'rippled', u'rippled_classic'], []], (u'clang.profile.nounity',): [[u'rippled', u'rippled_unity'], []], (u'clang.release.nounity',): [[u'rippled', u'rippled_unity'], []], (u'gcc.profile.nounity',): [[u'rippled', u'rippled_unity'], []], (u'clang.debug',): [[u'rippled', u'rippled_classic'], []], (u'gcc.release.nounity',): [[u'rippled', u'rippled_unity'], []], (u'clang.coverage.nounity',): [[u'rippled', u'rippled_unity'], []], (u'clang.debug.nounity',): [[u'rippled', u'rippled_unity'], []]}

CMAKE_ALL_GENERATE_OPTIONS: [(u'-Dassert=true', u'-Dsan=address'), (u'-GNinja',), (u'-Dassert=true',), (u'-GNinja', u'-Dstatic=true', u'-Dassert=true', u'-Dsan=thread'), (u'-Dsan=thread',), (u'-Dstatic=true', u'-Dassert=true'), (u'-GNinja', u'-Dstatic=true', u'-Dassert=true'), (u'-GNinja', u'-Dstatic=true', u'-Dsan=thread'), (u'-GNinja', u'-Dstatic=true', u'-Dsan=address'), (u'-Dstatic=true', u'-Dassert=true', u'-Dsan=address'), (u'-GNinja', u'-Dstatic=true', u'-Dassert=true', u'-Dsan=address'), (u'-GNinja', u'-Dsan=thread'), (u'-Dstatic=true', u'-Dassert=true', u'-Dsan=thread'), (u'-Dstatic=true', u'-Dsan=thread'), (u'-Dstatic=true', u'-Dsan=address'), (u'-GNinja', u'-Dsan=address'), (u'-GNinja', u'-Dstatic=true'), (u'-GNinja', u'-Dassert=true', u'-Dsan=thread'), (u'-GNinja', u'-Dassert=true', u'-Dsan=address'), (), (u'-Dassert=true', u'-Dsan=thread'), (u'-Dstatic=true',), (u'-GNinja', u'-Dassert=true'), (u'-Dsan=address',)]
```

**Comment by MarkusTeufelberger** [comment]
Thanks! :-)

**Comment by bachase** [comment]
@MarkusTeufelberger can you confirm 0.80.1 addresses this for now?

**Comment by MarkusTeufelberger** [comment]
At least starts the build on Arch and other Python 3 only distros now (https://github.com/MarkusTeufelberger/rippled-distrotest), so I'd call it a success. :+1: 

When I get around to it, I'll still try the refactor (not too happy about this whole "create a folder and another folder and clean up only after a bunch of builds" part), I'll just do a separate PR for that though.
